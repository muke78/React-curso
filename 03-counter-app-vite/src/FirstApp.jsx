// Manera normal de encapsulamiento de componentes sin ocupar div

import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <code> {JSON.stringify(newMessage)} </code> */}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};
// PropTypes
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  name: PropTypes.string,
};

/**
 * Las default props entras primero que las propTypes 
 * (compatibilidad con defaultProps se eliminará de los componentes 
 * de funciones en una versión importante futura.) */

// FirstApp.defaultProps = {
//   name: "Erick Gonzalez",
//   subtitle: "No hay subtitulo",
//   // title: "No hay ningun titulo",
// };

