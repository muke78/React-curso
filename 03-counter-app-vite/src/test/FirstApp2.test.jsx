import { render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe("Pruebas en <FirstApp/>", () => {
  const title = "Hola, soy Erick";
  const subtitle = "Hola soy un subtitulo";

  test("Debe de hacer la prueba con el snapshot", () => {
    const { container } = render(
      <FirstApp title={title} subtitle={subtitle} />
    );
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el mensaje "Hola, soy Erick"', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Debe de mostrar el titulo en un h1", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("Debe de mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(3);
  });
});

/**
 * El metodo toMatchSnapshot se encarga de tomar una muestra
 * del codigo original creando un componmente de html basico
 * que lo almacenara en una carpeta nueva dentro de donde
 * estamos haciendo las pruebas __snapshots__ con un archivo
 * FirstApp.test.jsx.snap, al momento de hascer modificaciones
 * en el codigo original, dara un error, porque la captura que
 * hay no fue igual a lo nuevo que el llego
 */

/**
 * toBeTruthy que se encuntre lo que se esta buscando
 *
 * toContain compara el contenido que se encuntra dentro
 * de una estructura de datos o de esquema sin importar
 * espacios o algo que influya en su contenido
 */

/**
 * Ver lo que se esta renderizando en el componente
 * screen.debug();
 */
