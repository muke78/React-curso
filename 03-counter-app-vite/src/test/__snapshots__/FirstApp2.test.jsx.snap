// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe de hacer la prueba con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Erick
  </h1>
  <p>
    Hola soy un subtitulo
  </p>
  <p>
    Hola soy un subtitulo
  </p>
  <p>
    Hola soy un subtitulo
  </p>
  <p />
</div>
`;
